packages/web/src/common/store/upload/sagaHelpers.ts

```diff
@@ -97,41 +98,50 @@ export function* recordGatedTracks(tracks: (TrackForUpload | TrackMetadata)[]) {
   yield* all(events.map((e) => put(e)))
 }
 
+export function* getUSDCMetadata(stream_conditions: USDCPurchaseConditions) {
+  const ownerAccount = yield* select(getAccountUser)
+  const wallet = ownerAccount?.erc_wallet ?? ownerAccount?.wallet
+  const ownerUserbank = yield* call(getUSDCUserBank, wallet)
+  const priceCents = stream_conditions.usdc_purchase.price
+  const priceWei = new BN(priceCents).mul(BN_USDC_CENT_WEI).toNumber()
+  const conditionsWithMetadata: USDCPurchaseConditions = {
+    usdc_purchase: {
+      ...stream_conditions.usdc_purchase,
+      price: priceCents,
+      splits: {
+        [ownerUserbank?.toString() ?? '']: priceWei
+      }
+    }
+  }
+  return conditionsWithMetadata
+}
+
 /**
+ * Adds relevant premium metadata
  * Converts prices to WEI and adds splits for USDC purchasable content.
  */
-export function* processTrackForUpload<T extends TrackMetadata>(track: T) {
+export function* addPremiumMetadata<T extends TrackMetadata>(
+  track: T // track or collection
```

Comment:
if it's track or collection, should we update the type? We could just make an interface or use the `PurchaseableContentMetadata` types

Reply:
I'm not seeing where we call this with collections anymore though. Can we just remove the comment?

Reply:
Ah yeah this is a leftover comment 🔪